python/pyraphtory/pyraphtory/context.py

```diff
@@ -101,6 +106,11 @@ def __init__(self, logging: bool = False):
         jars = ":".join([str(jar) for jar in jar_location.glob(env_jar_glob_lookup)])
         java_args = os.environ.get("PYRAPTHORY_JVM_ARGS", "")
 
+        # if jars is empty, then download it
+        if not jars:
+            jars = fileutils.download_raphtory(VERSION,
+                                               str(Path(inspect.getfile(self.__class__)).parent.parent / 'lib'))
+
```

Comment:
- Should we prompt the user before downloading the jar?
- Also, if the PYRAPHTORY_JAR_LOCATION was set, we probably should be raising an error if the jar is not   found rather than trying to download it?
- It does not seem like we check the version of the jar? Seems like we should check that the version is consistent by default? Also, does this mean that we would keep using the old jar if the python version is updated?

Reply:
Also noticed there is a bug on line 79-80 (this one is my fault I guess). These two lines should not be included in the `if logging` bit (i.e., the `logging` toggle should only disable the writing of the output, we still need the iterator to actually return the value to find the connection details. Can you fix this or should I do it?